### ISSUE TITLE
Filter Chat Messages

### ISSUE THREAD

**Issue by Feathers8** [Vencord/plugin-requests#159]
### Plugin Description

If someone sends a message containing a word you set, it would get blocked and not show on your screen nor give an unread message indicator. Messages in which you got pinged and DM's should probably bypass the filter (option toggle?).
It would be neat if it could also support Regex.

### Existing Plugin for other mod

https://betterdiscord.app/plugin/ChatFilter

### Request Agreement

- [X] I read the above and made sure my plugin is possible and allowed
- [X] I checked the [existing plugins](https://vencord.dev/plugins) and made sure my plugin doesn't already exist
- [X] I [searched the existing issues](https://github.com/Vencord/plugin-requests/issues?q=is%3Aissue) and made sure a similar request doesn't already exist

**Comment by EdVraz** [comment]
https://github.com/Vendicated/Vencord/pull/2238

**Comment by TrueReplayJay** [comment]
Yes please yes this. Ideally I would like censoring/replacement for words too like that plugin. It seems part of it has been addressed in https://github.com/Vendicated/Vencord/pull/2238 like the other user said.
